File: app/src/main/java/kolar/simplealarm/Model/AlarmReceiver.java
package kolar.simplealarm.Model;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import kolar.simplealarm.View.Activity.AlarmActivity;

import static kolar.simplealarm.Model.Controller.INTENT_ALARMCLASS;

/**
 * Created by Kolar on 26.11.2017.
 */

public class AlarmReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        context.startService(new Intent(context, AlarmSoundService.class));
        Intent i = new Intent(context, AlarmActivity.class);
        i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        i.putExtra(INTENT_ALARMCLASS, intent.getBundleExtra(INTENT_ALARMCLASS));
        context.startActivity(i);
    }
}

File: app/src/main/java/kolar/simplealarm/Model/AlarmReceiver.kt
package kolar.simplealarm.Model

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import kolar.simplealarm.Model.Controller.INTENT_ALARMCLASS
import kolar.simplealarm.View.Activity.AlarmActivity

/**
 * Created by ondrej.kolar on 22.01.2018.
 */
class AlarmReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        context?.let {
            intent?.let {
//                context.startService(Intent(context, AlarmSoundService::class.java))
                val i = Intent(context, AlarmActivity::class.java)
                i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                i.putExtra(INTENT_ALARMCLASS, intent.getBundleExtra(INTENT_ALARMCLASS))
                context.startActivity(i)
            }
        }
    }
}